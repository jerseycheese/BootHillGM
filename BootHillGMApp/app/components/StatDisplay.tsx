import React from 'react';

/**
 * StatDisplay component displays a single stat with label, value, and optional max value and description.
 */

interface StatDisplayProps {
  label: string; // The label of the stat (e.g., "Strength")
  value: number; // The current value of the stat
  max?: number; // The maximum value of the stat (optional)
  description?: string; // A description of the stat (optional)
}

export default function StatDisplay({ label, value, max, description }: StatDisplayProps) {
  return (
    <div id="bhgmStatDisplay" data-testid="stat-display" className="stat-container flex flex-col gap-2 bhgm-stat-display">
      <div className="stat-label text-sm font-bold">{label}</div>
      <div className="stat-value text-lg">
        {value}
        {max && (
          <span className="stat-max text-sm">
            / {max}
          </span>
        )}
      </div>
      {description && (
        <div className="stat-description text-xs italic">{description}</div>
      )}
    </div>
  );
}
