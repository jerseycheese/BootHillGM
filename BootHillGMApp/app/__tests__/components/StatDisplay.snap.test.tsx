import React from 'react';
import { render } from '@testing-library/react';
import StatDisplay from '@/app/components/StatDisplay';

describe('StatDisplay snapshots', () => {
  it('matches snapshot with basic props', () => {
    const { container } = render(<StatDisplay label="Strength" value={10} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with max value', () => {
    const { container } = render(<StatDisplay label="Health" value={8} max={10} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with description', () => {
    const { container } = render(
      <StatDisplay 
        label="Speed" 
        value={6} 
        description="Affects reaction time and movement"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with all props', () => {
    const { container } = render(
      <StatDisplay 
        label="Luck" 
        value={3} 
        max={7}
        description="Influences critical hits and random events"
      />
    );
    expect(container).toMatchSnapshot();
  });

  // Edge cases
  it('matches snapshot with zero value', () => {
    const { container } = render(<StatDisplay label="Empty Stat" value={0} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with large values', () => {
    const { container } = render(<StatDisplay label="Damage" value={9999} max={10000} />);
    expect(container).toMatchSnapshot();
  });

  // Testing specific elements
  it('renders with accessible attributes', () => {
    const { getByTestId } = render(
      <StatDisplay label="Accessibility" value={5} />
    );
    const statDisplay = getByTestId('stat-display');
    expect(statDisplay).toBeInTheDocument();
    expect(statDisplay).toMatchSnapshot();
  });
});
