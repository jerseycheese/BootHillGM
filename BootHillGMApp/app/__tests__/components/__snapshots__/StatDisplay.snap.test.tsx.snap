// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatDisplay snapshots matches snapshot with all props 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Luck
    </div>
    <div
      class="stat-value text-lg"
    >
      3
      <span
        class="stat-max text-sm"
      >
        / 
        7
      </span>
    </div>
    <div
      class="stat-description text-xs italic"
    >
      Influences critical hits and random events
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots matches snapshot with basic props 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Strength
    </div>
    <div
      class="stat-value text-lg"
    >
      10
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots matches snapshot with description 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Speed
    </div>
    <div
      class="stat-value text-lg"
    >
      6
    </div>
    <div
      class="stat-description text-xs italic"
    >
      Affects reaction time and movement
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots matches snapshot with large values 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Damage
    </div>
    <div
      class="stat-value text-lg"
    >
      9999
      <span
        class="stat-max text-sm"
      >
        / 
        10000
      </span>
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots matches snapshot with max value 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Health
    </div>
    <div
      class="stat-value text-lg"
    >
      8
      <span
        class="stat-max text-sm"
      >
        / 
        10
      </span>
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots matches snapshot with zero value 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-stat-display"
    data-testid="stat-display"
    id="bhgmStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Empty Stat
    </div>
    <div
      class="stat-value text-lg"
    >
      0
    </div>
  </div>
</div>
`;

exports[`StatDisplay snapshots renders with accessible attributes 1`] = `
<div
  class="stat-container flex flex-col gap-2 bhgm-stat-display"
  data-testid="stat-display"
  id="bhgmStatDisplay"
>
  <div
    class="stat-label text-sm font-bold"
  >
    Accessibility
  </div>
  <div
    class="stat-value text-lg"
  >
    5
  </div>
</div>
`;
